src/main/webapp/app/exercises/programming/manage/services/programming-exercise.service.ts

```diff
@@ -48,12 +49,25 @@ export class ProgrammingExerciseService {
         return this.http.put<string>(this.resourceUrl + '/' + exerciseId + '/generate-tests', { responseType: 'text' });
     }
 
+    /**
+     * Check plagiarism with JPlag
+     *
+     * @param exerciseId
+     */
+    checkPlagiarism(exerciseId: number): Observable<TextPlagiarismResult> {
+        return this.http
+            .get<TextPlagiarismResult>(`${this.resourceUrl}/${exerciseId}/check-plagiarism`, {
+                observe: 'response',
+            })
+            .pipe(map((response: HttpResponse<TextPlagiarismResult>) => response.body!));
+    }
+
     /**
      * Check for plagiarism
      * @param exerciseId of the programming exercise
      */
-    checkPlagiarism(exerciseId: number): Observable<HttpResponse<Blob>> {
-        return this.http.get(`${this.resourceUrl}/${exerciseId}/check-plagiarism`, {
+    checkPlagiarismJPlagResponse(exerciseId: number): Observable<HttpResponse<Blob>> {
```

Comment:
The title is a bit confusing, maybe you can use `checkPlagiarismJplagReport` instead?

Reply:
Good point, I actually intended to call it `checkPlagiarismJplagReport` but was probably a little distracted writing this 😄